packages/astro/src/core/server-islands/vite-plugin-server-islands.ts

```diff
@@ -37,6 +37,20 @@ export function vitePluginServerIslands({ settings }: { settings: AstroSettings
 					if (astro?.serverComponents.length) {
 						for (const comp of astro.serverComponents) {
 							if (!settings.serverIslandNameMap.has(comp.resolvedPath)) {
+								if (!settings.adapter) {
+									logger.error(
+										'islands',
+										"You tried to use a server island without an adapter. This an error and your project won't build.",
```

Comment:
```suggestion
										"You tried to render a server island without an adapter added to your project. An adapter is required to use the `server: defer` attribute on any component. Your project will fail to build unless you add an adapter or remove the attribute.",
```
Here's a suggestion for what this could be!

If they are getting an error message, we probably don't have to *say* this is an error. (We *might* not need to tell people it won't build, either. But I understand maybe wanting to say that if something will work in dev but only fail at build time.)